core/renderers/block_rendering_rewrite/block_render_draw.js

```diff
@@ -83,6 +84,17 @@ Blockly.blockRendering.Drawer.prototype.recordSizeOnBlock_ = function() {
   this.block_.startHat_ = this.info_.topRow.startHat;
 };
 
+/**
+ * Hide icons that were marked as hidden.
+ * @private
+ */
+Blockly.blockRendering.Drawer.prototype.hideHiddenIcons_ = function() {
+  for (var i = 0; i < this.info_.hiddenIcons.length; i++) {
+    var iconInfo = this.info_.hiddenIcons[i];
+    iconInfo.icon.iconGroup_.setAttribute('display', 'none');
```

Comment:
Where do you unhide a previously hidden icon?

Reply:
When an icon is not hidden, it is processed in layoutField which sets the display to block (which unhides it).